google/cloud/storage/blob.py

```diff
@@ -4797,3 +4809,126 @@ def _add_query_parameters(base_url, name_value_pairs):
     query = parse_qsl(query)
     query.extend(name_value_pairs)
     return urlunsplit((scheme, netloc, path, urlencode(query), frag))
+
+
+class Retention(dict):
+    """Map an object's retention configuration.
+
+    :type blob: :class:`Blob`
```

Comment:
Do we use this model of a separate class that links to a Blob in this way for any other classes?

Reply:
Not to Blob, but we use this same model in [Bucket for `IAMConfiguration` ](https://github.com/googleapis/python-storage/blob/9e4d1d8b44e55a1a10032b844932a8f5140c68fe/google/cloud/storage/bucket.py#L2212-L2220)

Reply:
Reading through the prd and [client request](https://goto.google.com/gcs-object-retention-client-request), I think it makes more sense to use a dataclass for the retention object. IMO, it will also provide more flexibility for further expansions and changes (which I have a feeling there may be in the future) Thoughts?

Reply:
Discussed offline and we decided to move forward with this dataclass implementation that is align with Bucket IAMConfiguration. This also gives us flexibility to expand on the retention object if/when new fields are introduced.